docs/developer/developer-fundamentals.md

```diff
@@ -0,0 +1,249 @@
+# Developer Fundamentals
+
+## Table of contents
+- [Developer Level-1](#developer-fundamentals)
+  - [Table of contents](#table-of-contents)
+  - [Importance of reading the material](#importance-of-reading-the-material)
+  - [Naming Conventions](#naming-conventions)
+    - [Branch Naming](#branch-naming)
+    - [File And Folder Naming](#file-and-folder-naming)
+    - [Class And Function Naming](#class-and-function-naming)
+    - [Variable Naming](#variable-naming)
+  - [Dividing The Code Into Commits](#dividing-the-code-into-commits)
+    - [Commit Too Big](#commit-too-big)
+      - [Bad Example:](#bad-example)
+    - [Commit Dependencies in the Correct Order](#commit-dependencies-in-the-correct-order)
+    - [Refactoring: Duplicate Commit](#refactoring-duplicate-commit)
+  - [Imitate Existing Solutions](#imitate-existing-solutions)
+  - [Simplify Complex Expressions](#simplify-complex-expressions)
+  - [Delete Unused Material](#delete-unused-material)
+  - [Add Patch Note](#add-patch-note)
+  - [Don't Hoard Issues](#dont-hoard-issues)
+  - [Don't Use Magic Numbers](#dont-use-magic-numbers)
+  - [Avoid Code Repetition](#avoid-code-repetition)
+  - [Avoid Tight Coupling](#avoid-tight-coupling)
+  - [Iterate Before Opening a PR](#iterate-before-opening-a-pr)
+  - [Resolving Conversations](#resolving-conversations)
+  - [When to Merge PR](#when-to-merge-pr)
+
+## Importance of reading the material
+
+-  Most people who skim through the reading materials tend to fail the trial. The key to passing is going through both [Developer Fundamentals](#developer-fundamentals) & [Developer Intermediate](./developer-intermediate.md) during each PR for the first 2 weeks.
```

Comment:
```suggestion
-  Most people who skim through the reading materials tend to fail at becoming long term contributors. The key to passing is going through the developer docs during each PR for the first 2 weeks. And coming back to them once a month to realign.
```

Reply:
Change it to something like this. 
There is a forgetting curve. People need to come back in certain intervals.